### ISSUE TITLE
[BUG] SOF linker error, .bss cannot fit into sof_fw section

### ISSUE THREAD

**Issue by aiChaoSONG** [thesofproject/sof#4491]
The APL FW cannot be built with XCC because of a linker error, the issue is also reproduced on CI.
```
/home/chao/xtensa/XtDevTools/install/tools/RG-2017.8-linux/XtensaTools/bin/xt-ld: sof section `.bss' will not fit in region `sof_fw'
make[3]: *** [CMakeFiles/sof.dir/build.make:1672: sof] Error 2
make[2]: *** [CMakeFiles/Makefile2:1415: CMakeFiles/sof.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:2240: src/arch/xtensa/CMakeFiles/bin.dir/rule] Error 2
make: *** [Makefile:736: bin] Error 2

```

**Comment by aiChaoSONG** [comment]
revert 38f18bd082747c7fc0859987c9a034c663118266 fix the issue

**Comment by slawblauciak** [comment]
P1 as it blocks SOF CI

**Comment by kkarask** [comment]
It affects CI SOF regression tests. Builds are failing due to that issue.

**Comment by lgirdwood** [comment]
@aiChaoSONG PR revert merged. Lets re-test.
@keyonjie fyi, will need a fix on APL.

**Comment by slawblauciak** [comment]
@aiChaoSONG good to close now?

**Comment by aiChaoSONG** [comment]
keyon provided new fix, 424beb0f8ff9548efeef8f2bc81baf70e0415e40, close.
